File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int size=nums.size();
        unordered_map<int,int> m;
        for(int i=0;i<size;i++){
            int x=nums[i];
            int y=target-x;
            if(m.find(y)!=m.end()){
                return {m[y],i};
            }
            m[nums[i]]=i;
        }
        return {-1,-1};
     }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map();
    for(let i = 0; i < nums.length; i++) {
        let x = nums[i];
        let y = target - nums[i];
        if (map.has(y)) {
            return [map.get(y), i];
        }
        map.set(x,i);
    }
};